File: 3471-MinimumAverageOfSmallestAndLargestElements/3471-MinimumAverageOfSmallestAndLargestElements.js
// Last updated: 10/03/2026, 10:02:24
/**
 * @param {number[]} nums
 * @return {number}
 */
var minimumAverage = function(nums) {
    nums.sort((a,b)=>a-b)
    let min=0;
    let max=nums.length-1;
    let minavg=Infinity;

    while(min<max){
        let avg=(nums[min]+nums[max])/2;
        minavg=Math.min(minavg,avg);
        min++;
        max--;
    }
    return minavg;
};

File: 3471-MinimumAverageOfSmallestAndLargestElements/3471-MinimumAverageOfSmallestAndLargestElements.cpp
// Last updated: 03/08/2026, 10:05:55
class Solution {
public:
    double minimumAverage(vector<int>& nums) {

        sort(nums.begin(), nums.end());

        double ans = INT_MAX;

        int left = 0;
        int right = nums.size() - 1;

        while (left < right) {

            double avg = (nums[left] + nums[right]) / 2.0;

            ans = min(ans, avg);

            left++;
            right--;
        }

        return ans;
    }
};